dispatchers/libdispatcher.go

```diff
@@ -120,3 +160,105 @@ func (d *RoundRobinDispatcher) ConnIDs() (connIDs []string) {
+type BroadcastDispatcher struct {
+	sync.RWMutex
+	conns    engine.DispatcherConns
+	strategy StrategyDispatcher
+}
+
+func (d *BroadcastDispatcher) SetProfile(pfl *engine.DispatcherProfile) {
+	d.Lock()
+	pfl.Conns.Sort()
+	d.conns = pfl.Conns.Clone() // avoid concurrency on profile
+	d.Unlock()
+	return
+}
+
+func (d *BroadcastDispatcher) ConnIDs() (connIDs []string) {
+	d.RLock()
+	connIDs = d.conns.ConnIDs()
+	d.RUnlock()
+	return
+}
+
+func (d *BroadcastDispatcher) Dispatch(conns map[string]*rpcclient.RpcClientPool, routeID *string,
+	serviceMethod string, args interface{}, reply interface{}) (lastErr error) { // no cache needed for this strategy because we need to call all connections
+	return d.strategy.Dispatch(d.ConnIDs(), conns, routeID, serviceMethod, args, reply)
+}
+
+func (ign *singleResultStrategyDispatcher) Dispatch(connIDs []string, conns map[string]*rpcclient.RpcClientPool, routeID *string,
+	serviceMethod string, args interface{}, reply interface{}) (err error) {
+	var connID string
+	if routeID != nil &&
+		*routeID != "" {
+		// use previously discovered route
+		if x, ok := engine.Cache.Get(utils.CacheDispatcherRoutes,
+			*routeID); ok && x != nil {
+			connID = x.(string)
+			if err = conns[connID].Call(serviceMethod, args, reply); !utils.IsNetworkError(err) {
+				return
+			}
+		}
+	}
+	for _, connID = range connIDs {
+		conn, has := conns[connID]
+		if !has {
+			err = utils.NewErrDispatcherS(
+				fmt.Errorf("no connection with id: <%s>", connID))
+			continue
+		}
+		if err = conn.Call(serviceMethod, args, reply); utils.IsNetworkError(err) {
+			continue
+		}
+		if routeID != nil &&
+			*routeID != "" { // cache the discovered route
+			engine.Cache.Set(utils.CacheDispatcherRoutes, *routeID, connID,
+				nil, true, utils.EmptyString)
+		}
+		break
+	}
+	return
+}
+
+func (ign *brodcastStrategyDispatcher) Dispatch(connIDs []string, conns map[string]*rpcclient.RpcClientPool, routeID *string,
+	serviceMethod string, args interface{}, reply interface{}) (err error) {
+	var hasErrors bool
+	var hasOnlyErrors bool = true
+	for _, connID := range connIDs {
+		conn, has := conns[connID]
+		if !has {
+			err = utils.NewErrDispatcherS(
+				fmt.Errorf("no connection with id: <%s>", connID))
+			utils.Logger.Err(fmt.Sprintf("<%s> Error at %s strategy for connID %q : %s",
+				utils.DispatcherS, utils.MetaBroadcast, connID, err.Error()))
+			hasErrors = true
+			continue
+		}
+		if err = conn.Call(serviceMethod, args, reply); utils.IsNetworkError(err) {
+			utils.Logger.Err(fmt.Sprintf("<%s> Network Error at %s strategy for connID %q : %s",
+				utils.DispatcherS, utils.MetaBroadcast, connID, err.Error()))
+			hasErrors = true
+			continue
+		} else if err != nil {
+			utils.Logger.Err(fmt.Sprintf("<%s> Error at %s strategy for connID %q : %s",
+				utils.DispatcherS, utils.MetaBroadcast, connID, err.Error()))
+			hasErrors = true
+			continue
+		}
+		hasOnlyErrors = false
+	}
+	if hasOnlyErrors {
```

Comment:
Please remove hasOnlyErrors since it does not help much (if one error, the whole cache reload is unreliable).